modules/light-clients/07-tendermint/update.go

```diff
@@ -137,7 +137,10 @@ func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, client
 		panic(fmt.Errorf("expected type %T, got %T", &Header{}, clientMsg))
 	}
 
-	cs.pruneOldestConsensusState(ctx, cdc, clientStore)
+	// don't do prune logic in CheckTx
+	if !ctx.IsCheckTx() {
+		cs.pruneOldestConsensusState(ctx, cdc, clientStore)
+	}
```

Comment:
if we are avoiding this in `CheckTx` then should it also be avoided in `ReCheckTx`?

Is there something that is particularly expensive about this other than just wanting to avoid unnecessary store operations in CheckTx? I know we construct an iterator but it only used to access the oldest header info in store and breaks immediately after the first cb.

Reply:
Oh no it should have been both, great catch!

Reply:
Is this just because you've suggested in the issue that we should avoid client updates on ReCheckTx completely?